aiida_common_workflows/workflows/relax/bigdft/generator.py

```diff
@@ -108,21 +186,78 @@ def get_builder(
             **kwargs
         )
 
+        builder = self.process_class.get_builder()
+
         if relax_type == RelaxType.ATOMS:
             relaxation_schema = 'relax'
+        elif relax_type == RelaxType.NONE:
+            relaxation_schema = 'relax'
+            builder.relax.perform = orm.Bool(False)
         else:
             raise ValueError('relaxation type `{}` is not supported'.format(relax_type.value))
 
-        builder = self.process_class.get_builder()
-        builder.structure = structure
+        pymatgen_struct = structure.get_pymatgen()
+        ortho_dict = None
+        # for single atom computations, we should actually not perform relaxation
+        if pymatgen_struct.ntypesp <= 1:
+            builder.relax.perform = orm.Bool(False)
+            # pass the structure through a transform to generate orthorhombic structure if possible/needed.
+            newstruct, ortho_dict = ortho_struct(structure)
+            newstruct.store()
```

Comment:
There are two problems with this approach:
1. You are losing the provenance of the new structure here
2. The `get_builder` call will now create new nodes in the database, even if the builder is discarded afterwards

The fix to the first one is simple: just turn `ortho_struct` into a `calcfunction`. The second one is more tricky. We also face this dilemma in `aiida-quantumespresso` but I am not sure that it is a good idea to have the `get_builder` create nodes in the database. As I user I would not expect this and feel like the `get_builder` should be a no-op from the database' perspective.

Can you explain what this is doing and why it is necessary? If it is always necessary, would it make sense to integrate it in the BigDFT workchain as a first step in the outline?  Then there is no problem with provenance being added in the builder setup and all is nicely wrapped up in the workchain.

Reply:
Making it a calcfunction seems perfectly fine.
This is to handle the specific case of non orthorhombic structures and to try to generate an orthorhombic supercell if possible. As it potentially adds sites to the structure, this should be done before calling functions that will take the number of atoms as an input to generate inputfile parameters, so I don't think we could move it inside the workchain easily for now. 
A possibility would be to call it once in the builder, without storing the result in the db, use it to generate the input, and then call it once again inside the workchain with proper handling of the output? A bit hacky, but it may work